src/CommonDCModelDropdown.php

```diff
@@ -212,14 +212,130 @@ public static function getSpecificValueToDisplay($field, $values, array $options
+                $orientation = $item_rack['orientation'];
+                $hpos = $input['is_half_rack'] ?? $this->fields['is_half_rack'] ? $item_rack['hpos'] : Rack::POS_NONE;
+                $depth = $input['depth'] ?? $this->fields['depth'];
+
+                for ($i = 0; $i < $requiredUnits; $i++) {
+                    $position = $item_rack['position'] + $i;
+
+                    if (isset($filled[$position]) && $this->checkCellFilled($filled[$position], $orientation, $hpos, $depth)) {
+                        $errors = [
+                            sprintf(
+                                __(
+                                    'Unable to update model because it is used by an asset in a rack %s and the new required units do not fit into the rack'
+                                ),
+                                $rack->getLink()
+                            )
+                        ];
+                        break;
+                    }
+
+                    if ($position > $rack->fields['number_units']) {
+                        for ($j = 1; $j <= $requiredUnits - $i; $j++) {
+                            $position = $item_rack['position'] - $j;
+                            if (
+                                isset($filled[$position]) && $this->checkCellFilled($filled[$position], $orientation, $hpos, $depth)
+                                || $position < 1
+                            ) {
+                                $errors = [
+                                    sprintf(
+                                        __(
+                                            'Unable to update model because it is used by an asset in a rack %s and the new required units do not fit into the rack'
+                                        ),
+                                        $rack->getLink()
+                                    )
+                                ];
+                                break 2;
+                            }
+                        }
+                        break;
+                    }
+                }
+            }
+
+            if (!empty($errors)) {
+                foreach ($errors as $error) {
+                    Session::addMessageAfterRedirect(
+                        $error,
+                        true,
+                        ERROR
+                    );
+                }
+                return false;
+            }
+        }
+
+        return $input;
+    }
+
+    public function post_updateItem($history = 1)
+    {
+        if (!$this->fields['is_half_rack']) {
+            $item_rack = new Item_Rack();
+            foreach ($this->getItemRackForModel() as $item) {
+                if ($item['hpos'] == Rack::POS_NONE) {
+                    continue;
+                }
+
+                $item_rack->update([
+                    'id' => $item['id'],
+                    'hpos' => Rack::POS_NONE,
+                ]);
+            }
+        }
```

Comment:
Could you add a small code comment to indicate that we are doing this because a full rack model can't use an half space ?

Also maybe the opposite check is also needed ? An half-rack model should always use an half space (move it to `POS_NONE` -> `POS_LEFT` ?) 

Unless I'm mistaken this code does not seem to be tested, could you add some tests ?

Reply:
Also could be isolated into a method with a fitting name :)